### PR TITLE
Simplify Tools

### PR DESCRIPTION
**Pull request by 0xc0170**
As the comment says:

+    # Tools dictionary
+    # Each of this tool needs to support at least:
+    # - get_toolchain (toolchain is a list of toolchains supported by tool)
+    # - get_toolname (returns name string)
+    # - export_project (basic functionality to be covered by a tool)

### CODE REVIEW

File: project_generator/project.py [hunk 1]
```diff
@@ -391,19 +400,23 @@ def build(self, tool):
         else:
             tools = [tool]
 
+        result = 0
+
         for build_tool in tools:
-            builder = self.tools.get_value(build_tool, 'tool')
+            builder = self.tools.get_tool(build_tool)
```
Comment (sarahmarshy) [inline project_generator/project.py]: more specific and clear +1

File: project_generator/project.py [hunk 2]
```diff
@@ -445,11 +458,11 @@ def format_source_files(self, ext, tool_specific_settings, toolchain_specific_se
                 k, v in settings.items()},toolchain_specific_settings.source_of_type(ext))]
 
     def customize_project_for_tool(self, tool):
-        toolchain_specific_settings =  self.tool_specific[self.tools.get_value(tool, 'toolchain')]
+        toolchain_specific_settings =  self.tool_specific[self.tools.get_toolchain(tool)]
         tool_specific_settings = []
-        toolnames = self.tools.get_value(tool, 'toolnames')
+        toolnames = self.tools.get_toolnames(tool)
```
Comment (sarahmarshy) [inline project_generator/project.py]: is self.tools ToolsSupported() too?

File: project_generator/project.py [hunk 3]
```diff
@@ -391,19 +400,23 @@ def build(self, tool):
...
 
             logging.debug("Building for tool: %s", build_tool)
             logging.debug(self.generated_files)
             builder(self.generated_files[build_tool], self.pgen_workspace.settings).build_project()
+            return result
 
     def get_generated_project_files(self, tool):
         # returns list of project files which were generated
-        exporter = ToolsSupported().get_value(tool, 'tool')
+        exporter = ToolsSupported().get_tool(tool)
```
Comment (sarahmarshy) [inline project_generator/project.py]: could use self.tools here?

Comment (sarahmarshy) [inline project_generator/project.py]: Or eliminate self.tools and always use ToolsSupported()

**Comment by 0xc0170** [comment]
All tests green. I started returning results from some of functions I touched, will continue with this effort.

File: project_generator/project.py (hunk 3 above)
Comment (0xc0170) [inline project_generator/project.py]: Going to eliminate self.tools

Comment (sarahmarshy) [inline project_generator/project.py]: I think that ToolsSupported() is more readable, sounds good

**Pull request merged**